### ISSUE TITLE
the connection self-closed just before a subscribe receive it second message

### ISSUE THREAD

**Issue by odlex** [nats-io/nats.net.v1#358]
I make an uwp game made with Unity.

**N.B. the problem only appear in uwp build not in a standard build where all work perfectly**

** The unity code of subscriber
```
try {
	EventHandler<ConnEventArgs> ClosedEventHandler = (sender, args) => {
		Debug.LogError("###########ClosedEventHandler");
	};
	EventHandler<ConnEventArgs> DisconnectedEventHandler = (sender, args) => {
		Debug.LogError("###########DisconnectedEventHandler");
	};

	var options = ConnectionFactory.GetDefaultOptions();
	options.UseOldRequestStyle = false;
	options.Verbose = false;
	options.NoEcho = false;
	options.Pedantic = false;
	options.Url = "nats://[::1]:4222";
	options.ClosedEventHandler = ClosedEventHandler;
	options.DisconnectedEventHandler = DisconnectedEventHandler;

	c = new ConnectionFactory().CreateConnection(options);
	IAsyncSubscription subscriber = c.SubscribeAsync(Subject, (sender, args) =>
	{
	   Debug.Log($"Subject: {args.Message.Subject} | PayLoad>: {Encoding.UTF8.GetString(args.Message.Data)}";e);
	});
} catch (Exception e) {
	Debug.LogError(e.Message);
	Debug.LogError(e.StackTrace);
}
```

** I have secondary application (only c# xaml UWP) who just publish on a specific subject when push a button
```
c.Publish(Subject, payload);
c.Flush()
```

**Find bellow the logs of nats server:**
```
[458480] 2020/01/24 12:55:22.235590 [DBG] ::1:56114 - cid:1878 - Client connection created
[458480] 2020/01/24 12:55:22.235590 [TRC] ::1:56114 - cid:1878 - ->> [CONNECT {"verbose":false,"pedantic":false,"user":"","pass":"","ssl_required":false,"name":"","auth_token":"","lang":".NET","version":"0.0.1","protocol":1,"jwt":"","nkey":"","sig":"","echo":true}]
[458480] 2020/01/24 12:55:22.236583 [TRC] ::1:56114 - cid:1878 - ->> [PING]
[458480] 2020/01/24 12:55:22.236583 [TRC] ::1:56114 - cid:1878 - <<- [PONG]
[458480] 2020/01/24 12:55:22.236583 [TRC] ::1:56114 - cid:1878 - ->> [SUB PinBallGroup  1]
[458480] 2020/01/24 12:55:22.236583 [TRC] ::1:56114 - cid:1878 - ->> [PING]
[458480] 2020/01/24 12:55:22.236583 [TRC] ::1:56114 - cid:1878 - <<- [PONG]
[458480] 2020/01/24 12:55:35.398895 [TRC] ::1:56114 - cid:1878 - ->> [PUB PinBallGroup 19]
[458480] 2020/01/24 12:55:35.398895 [TRC] ::1:56114 - cid:1878 - ->> MSG_PAYLOAD: [**Coucou > 50,85677]
[458480] 2020/01/24 12:55:35.398895 [TRC] ::1:55421 - cid:1601 - <<- [MSG PinBallGroup 1 19]
[458480] 2020/01/24 12:55:35.399895 [TRC] ::1:56114 - cid:1878 - <<- [MSG PinBallGroup 1 19]
[458480] 2020/01/24 12:55:35.399895 [DBG] ::1:56114 - cid:1878 - Client connection closed
[458480] 2020/01/24 12:55:35.452894 [DBG] ::1:56115 - cid:1879 - Client connection created
[458480] 2020/01/24 12:55:35.452894 [TRC] ::1:56115 - cid:1879 - ->> [CONNECT {"verbose":false,"pedantic":false,"user":"","pass":"","ssl_required":false,"name":"","auth_token":"","lang":".NET","version":"0.0.1","protocol":1,"jwt":"","nkey":"","sig":"","echo":true}]
[458480] 2020/01/24 12:55:35.453895 [TRC] ::1:56115 - cid:1879 - ->> [PING]
[458480] 2020/01/24 12:55:35.453895 [TRC] ::1:56115 - cid:1879 - <<- [PONG]
[458480] 2020/01/24 12:55:35.453895 [TRC] ::1:56115 - cid:1879 - ->> [SUB PinBallGroup  1]
[458480] 2020/01/24 12:55:35.453895 [TRC] ::1:56115 - cid:1879 - ->> [PING]
[458480] 2020/01/24 12:55:35.453895 [TRC] ::1:56115 - cid:1879 - <<- [PONG]
```

**Comment by ColinSullivan1** [comment]
Thank you for the debug/trace output from the server.  A little more information from the client will help us debug this.

Do you see the DisconnectedEventHandler being triggered?  If so can you print out `args.Error`?

Can you also add an Async error handler and print out the error?  Here's an example of setting up that handler:  https://github.com/nats-io/nats.net/blob/master/src/Samples/Benchmark/Benchmark.cs#L205

It looks like the connection is being terminated and the client is reconnecting, but we are just not sure what is triggering the client to disconnect.  

You have two subscribers, one one connection ID 1601 which receives the message and does not exit, and another with connection ID 1878, which does exit upon receiving the message.  Are both of those C# UPW clients?

**Comment by ColinSullivan1** [comment]
This is very likely related to issue #361.

**Comment by ColinSullivan1** [comment]
We believe this has been resolved by #370.  Please re-open this if you continue to encounter the issue.
